src/tutorials/0005/05.md

```diff
@@ -0,0 +1,69 @@
+```javascript
+await ipfs.add(file, { wrapWithDirectory: true }) // add a single file
+await ipfs.add([file1, file2], { wrapWithDirectory: true }) // add multiple files via an array
+```
+
+When we used the `add` method previously, we were able to pass in just a `File` object from our browser. However, to wrap files with a directory, we need to also provide a path with the name we want for each file. To do this, we must replace the `file` argument in the `add` function call with an object or an array of objects, one for each file, structured like so:
+
+```javascript
+{
+    path: 'file.txt', // a string for our desired path, including the filename
+    content: file // in our case, a File object
+}
+```
+
+To add multiple files at once, we just need to pass an array of objects like the one shown above to the `add` method. For example, we could add two cat pictures to a root directory on our IPFS node like so:
+
+```javascript
+await ipfs.add([
+    {
+        path: 'adorable-kitty.jpg',
+        content: catPic1
+    },
+    {
+        path: 'cat-drinking-milk.jpg',
+        content: catPic2
+    }
+], { wrapWithDirectory: true })
+```
+
+The `add` method will return information about all the files we've added, as well as details about any directory it had to create in order to correctly add the files in the correct paths. For example, in the case above, the `add` method would return us an array of three elements, one for each file added and another for the root directory created.
+
+Note that we could also have used the same `add` method to created one or more subdirectories by adding them to the path of each file. For example:
+
+```javascript
+await ipfs.add([
+    {
+        path: 'cats/adorable-kitty.jpg',
+        content: catPic1
+    },
+    {
+        path: 'cats/cat-drinking-milk.jpg',
+        content: catPic2
+    },
+    {
+        path: 'dogs/dog-on-a-table.jpg',
+        content: dogPic1
+    }
+], { wrapWithDirectory: true })
+```
+In this case we'd end up with six objects in our resulting array: one for each of the three files added, one for the new root directory, one for the new `cats` directory, and one for the new `dogs` directory.
+
+## Directories are immutable
+
+It's important to note that directories we create in this way do not behave as in a regular file system. If we've wrapped some files with a directory with using `add`, we can't simply `add` new files to that directory. The contents of the directory we just created are final and immutable.
+
+This is because of the way in which IPFS uses [content addressing](https://proto.school/#/data-structures/03): different contents lead to a different cryptograhpic hash (Content Identifier), whether we're working with directories or files themselves.
+
+But what if you forgot to add a file to a directory you just created? You'll have to call the `add` method again with all the files you want wrapped with that directory, resulting in a new directory altogether with a new CID.
```

Comment:
Love the idea of adding this point.

Reply:
Is this reasonably accurate @alanshaw? 

> But what if you forgot to add a file to a directory you just created? You'll have to call the `add` method again with all the files you want wrapped with that directory, resulting in a new directory altogether with a new CID.
> 
> However, it's important to note that the file won't actually be stored multiple times if you do this. Unlike on your computer, where you can accidentally download the same file twice, store it in two different directories, and double your storage needs, the IPFS network inherently knows the contents of a file based on its CID, and can therefore keep only a single copy of the data that the CID refers to. It's only the reference to that data (the CID, which you can think of as a link or an addresses), which is stored in each of the directories you create. This efficiency is one of IPFS's key advantages.
> 
> **Rather than thinking of directories created with `{ wrapWithDirectory: true }` as traditional file folders, it's more useful to think of them as naming shortcuts, as we'll see in upcoming lessons.**

Reply:
that's been added ^ please let us know if you think it needs edits